folly/net/NetOps.h

```diff
@@ -76,8 +76,6 @@ struct mmsghdr {
   struct msghdr msg_hdr;
   unsigned int msg_len;
 };
-#else
```

Comment:
I think we should change the test above to just `#if !FOLLY_HAVE_SENDMMSG`.

Reply:
This will cause issue for older version of Linux that doesnt have sendmmsg.  Code wont compile due to multiple define of mmsghdr:

net/NetOps.h:76:8: error: redefinition of 'struct mmsghdr'
 struct mmsghdr {
        ^~~~~~~
In file included from /usr/include/sys/socket.h:40:0,
                 from /usr/include/netinet/in.h:25,

Reply:
Do some versions of Linux have `mmsghdr` but not `sendmmsg`?

Reply:
The issue is that mmsghdr is used by both sendmmsg and recvmmsg.  recvmmsg was added to Linux 2.6.33 but sendmmsg was added to Linux 3.0.  Between those 2 versiosn, there is mmsghdr but no sendmmsg.

Reply:
Alright, let's do a cmake check for `recvmmsg` and a check for `sendmmsg`.

We will use the result of the check for `recvmmsg` in the test above conditionally defining `mmsghdr`, and the result of the check for `sendmmsg` for guarding calls to `sendmmsg`.

Reply:
sounds good, will add FOLLY_HAVE_RECVMMSG and update the if statement

Reply:
I have committed the changes.  Another thing that I have to add is to include Portability.h so folly-config.h can be included.

Reply:
Build failed.  But not sure if is related to any changes from this PR after went through the build log

Reply:
Build failure appears unrelated.